Spelling

jsoref opened the pull request:
### Context
Misspellings make it harder to search for things; they make it harder for people to understand code and descriptions; and sometimes they result in bugs.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

I'm happy to drop files/directories if they're not yours -- I can't figure out if `subprojects/` is third party content.
I'm happy to split into, e.g. API, docs, code as requested.

--
Generated by https://github.com/jsoref/spelling `f` -- this is the misspelled `a` family

In short, my tool identifies candidates, I manually select and inspect candidates, and corrections are chosen by me manually (although I often let Google suggest the correction instead of typing it myself as it has a lower typo rate).

This is part 3 (part one was #8147)

jsoref commented:
@blindpirate

blindpirate commented:
Thanks @jsoref , this is awesome! 

I reverted two commits because I think we should be careful when it comes to class name: `Deprecatable` and `XCTestExecuter`.

Let's wait for the CI feedback.

BTW, have you considered integrating the spelling check to some code quality plugin?

jsoref commented:
I've thought about it, but haven't figured out a good way to deal w/ it.

I've started adding support for Travis hooks (checkstyle and powerdns), the biggest problem is that my tooling doesn't know the difference between garbage and misspellings, so it only works well w/ a human. The hook works effectively as regression testing preventing the unintentional introduction of new misspellings.

This month, I started looking into bots for GitHub, which seems like an interesting approach, so I've been toying around the idea of trying to integrate that way, but I haven't figured out the details. One thing is that I don't have the resources to cover the CPU resources of hosting. w/ Travis integration, I'm just sharing the project's own CPU resources which they accept.

jsoref commented:
Oh, the other thing is that the code quality things tend not to be fans of Perl.

I'll probably have to port my code to either Java (checkstyle) or JS (probot or similar). Not a big deal.

I had a tool I once wrote in Perl that I rewrote later in Python (I need to find it and look into a hosting solution for it, after I figure out a solution for this tool). The logic for this spelling checker was originally written in JavaScript (xpcshell, long before Node.js) which I rewrote in Perl.

jsoref commented:
@blindpirate, that said, it looks like Travis and TeamCity are happy.
